src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -271,6 +271,8 @@ public class HttpSource : IDisposable
             try
             {
                 response = await RetryHandler.SendAsync(request, log, cancellationToken);
+				
+                response.LogServerWarning(log);
```

Comment:
Makes sense. I'm assuming I can retrieve the logger through a combination of https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSourceAuthenticationHandler.cs#L84 and https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSourceAuthenticationHandler.cs#L111?